### PR TITLE
Fix/asset formatters zero decimals

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
- Added strict checks for undefined values at `formatAssetAmountCurrency` and `formatAssetAmount`

### CODE REVIEW

**Review by junkai121 — APPROVED** [review]
LGTM!

File: packages/xchain-util/src/asset.test.ts [hunk 1]
```diff
@@ -115,6 +115,11 @@ describe('asset', () => {
       const amount = assetAmount(11.001, 8)
       expect(formatAssetAmount({ amount, decimal: 5, trimZeros: true })).toEqual('11.001')
     })
+
+    it('formats an `AssetAmount` with 0 decimal', () => {
+      const amount = assetAmount(11.001, 0)
+      expect(formatAssetAmount({ amount, decimal: 5, trimZeros: true })).toEqual('11')
```
Comment (veado) [inline packages/xchain-util/src/asset.test.ts:121]: Please add a test w/o using `trimZeros`

```
 expect(formatAssetAmount({ amount, decimal: 5 })).toEqual('11.00000')
 ```

File: packages/xchain-util/src/asset.test.ts [hunk 2]
```diff
@@ -297,6 +302,11 @@ describe('asset', () => {
       const amount = assetAmount(10.01, 8)
       expect(formatAssetAmountCurrency({ amount, asset: AssetBNB, trimZeros: true })).toEqual('10.01 (BNB)')
     })
+
+    it('formats amount by using 0 decimal of asset', () => {
+      const amount = assetAmount(10.01, 0)
+      expect(formatAssetAmountCurrency({ amount, asset: AssetBNB, trimZeros: true })).toEqual('10 (BNB)')
```
Comment (veado) [inline packages/xchain-util/src/asset.test.ts:308]: Same here, pls add a test w/o using trimZeros, but using `decimal` for `formatAssetAmountCurrency`

```
 expect(formatAssetAmountCurrency({ amount, asset: AssetBNB, decimal: 2,  trimZeros: false })).toEqual('10.00 (BNB)')
 ```

**Review by veado — APPROVED** [review]
LGTM! Just a suggestion for two more tests

**Review by veado — APPROVED** [review]

**Pull request merged**